public/app/features/connections/pages/EditDataSourcePage.tsx

```diff
@@ -4,18 +4,41 @@ import { useLocation, useParams } from 'react-router-dom';
 import { Page } from 'app/core/components/Page/Page';
 import { EditDataSource } from 'app/features/datasources/components/EditDataSource';
 import { EditDataSourceActions } from 'app/features/datasources/components/EditDataSourceActions';
-
-import { useDataSourceSettingsNav } from '../hooks/useDataSourceSettingsNav';
+import { EditDataSourceTitle } from 'app/features/datasources/components/EditDataSourceTitle';
+import { useDataSourceInfo } from 'app/features/datasources/components/useDataSourceInfo';
+import { useDataSourceSettingsNav } from 'app/features/datasources/state';
 
 export function EditDataSourcePage() {
   const { uid } = useParams<{ uid: string }>();
   const location = useLocation();
   const params = new URLSearchParams(location.search);
   const pageId = params.get('page');
-  const { navId, pageNav } = useDataSourceSettingsNav();
+  const nav = useDataSourceSettingsNav(uid, pageId);
+
+  const info = useDataSourceInfo({
+    dataSource: nav.dataSource,
+    dataSourcePluginName: nav.main.dataSourcePluginName,
+    isDefault: nav.dataSource.isDefault,
+    alertingSupported: nav.dataSourceHeader.alertingSupported,
+    onUpdate: nav.dataSourceHeader.onUpdate,
+  });
```

Comment:
instead of this abstraction, I would just make a wrapper component that owns setting up the Page component (and passes in navId as props). DataSourceTabPage or something (so we can easily have the same header for all the tabs)